sdk/servicebus/service-bus/src/track2/serviceBusReceiverClient.ts

```diff
@@ -0,0 +1,526 @@
+   */
+  streamMessages(
+    handler: MessageHandlers<ContextType<LockModeT>>,
+    options?: StreamMessagesOptions
+  ): void;
+
+  /**
+   * Returns an iterator that can be used to receive messages from Service Bus.
+   * @param options Options for iterateMessages.
+   */
+  iterateMessages(options?: IterateMessagesOptions): MessageIterator<ContextType<LockModeT>>;
+
+  /**
+   * Receives, at most, `maxMessages` worth of messages.
+   * @param maxMessages The maximum number of messages to accept.
+   * @param maxWaitTimeInSeconds The maximum time to wait, in seconds, for messages to arrive.
+   * @param options Options for receiveBatch.
+   */
+  receiveBatch(
+    maxMessages: number,
+    maxWaitTimeInSeconds?: number,
+    options?: ReceiveBatchOptions
+  ): Promise<Message[]>;
+
+  /**
+   * Closes the client.
+   */
+  close(): Promise<void>;
+
+  /**
+   * Methods related to service bus diagnostics.
+   */
+  diagnostics: {
+    /**
+     * Peek within a queue or subscription.
+     * NOTE: this method does not respect message locks or increment delivery count
+     * for messages.
+     * @param maxMessageCount The maximum number of messages to retrieve.
+     */
+    peek(maxMessageCount?: number): Promise<Message[]>;
+
+    /**
+     * Peek within a queue or subscription, starting with a specific sequence number.
+     * NOTE: this method does not respect message locks or increment delivery count
+     * for messages.
+     * @param fromSequenceNumber The sequence number to start peeking from (inclusive).
+     * @param maxMessageCount The maximum number of messages to retrieve.
+     */
+    peekBySequenceNumber(fromSequenceNumber: Long, maxMessageCount?: number): Promise<Message[]>;
+  };
+}
+
+/**
+ * Methods to manage rules for subscriptions. More information about subscription rules
+ * can be found here: https://docs.microsoft.com/en-us/azure/service-bus-messaging/topic-filters
+ */
+export interface SubscriptionRuleManagement {
+  getRules(): Promise<RuleDescription[]>;
+  removeRule(ruleName: string): Promise<void>;
+  addRule(
+    ruleName: string,
+    filter: boolean | string | CorrelationFilter,
+    sqlRuleActionExpression?: string
+  ): Promise<void>;
+}
+
+export type ClientTypeT<
+  ReceiveModeT extends "peekLock" | "receiveAndDelete",
+  EntityTypeT extends "queue" | "subscription",
+  SessionsEnabledT extends "sessions" | "nosessions"
+> = SessionsEnabledT extends "nosessions"
+  ? EntityTypeT extends "queue"
+    ? NonSessionReceiver<ReceiveModeT>
+    : NonSessionReceiver<ReceiveModeT> & SubscriptionRuleManagement
+  : EntityTypeT extends "queue"
+  ? SessionReceiver<ReceiveModeT>
+  : SessionReceiver<ReceiveModeT> & SubscriptionRuleManagement;
+
+export interface ServiceBusReceiverClient {
+  /**
+   * Creates a client for an Azure Service Bus queue.
+   *
+   * @param queueAuth Data needed to connect to a queue.
+   * @param receiveMode The receive mode to use (defaults to PeekLock)
```

Comment:
Oh interesting, so it is actually safe to have a default receive mode? I do agree that this makes sense as the default it we're going to have one.

Reply:
It's the safest option vs receiveAndDelete for sure - at least you don't lose messages but you might infinitely reprocess them if you don't realize you have to complete yourself. :)

Reply:
Isn't there some limit where if you receive a message X number of times and don't settle it, it ends up going to the dead-letter queue?

Reply:
Limit - `Max Delivery Count`, by default the value is 10, once the count reaches 10, the message is dropped in the dead-letter queue.